Adding an option to use a threadpool for Request/Connection processing.

NiteshKant opened the pull request:
If HTTP Request processing or TCP connection handling is blocking or is cpu intensive, there isn't a straightforward way to handoff this processing from the channel's eventloop to a different threadpool.

 This change adds this feature. The examples for TCP and HTTP are added to rx-netty-examples under cpuintensive package.

Commit:
Adding an option to use a threadpool for Request/Connection processing.

If HTTP Request processing or TCP connection handling is blocking or is cpu intensive, there isn't a straightforward way to handoff this processing from the channel's eventloop to a different threadpool.

 This change adds this feature. The examples for TCP and HTTP are added to rx-netty-examples under cpuintensive package.

cloudbees-pull-request-builder commented:
[RxNetty-pull-requests #118](https://netflixoss.ci.cloudbees.com/job/RxNetty-pull-requests/118/) FAILURE
Looks like there's a problem with this pull request

Commit:
ClientChannelFactoryImpl was not catching errors inside operationComplete()

This caused any errors not getting propagated to the connect() subscriber.

Also added some debug for a flaky test.

cloudbees-pull-request-builder commented:
[RxNetty-pull-requests #119](https://netflixoss.ci.cloudbees.com/job/RxNetty-pull-requests/119/) SUCCESS
This pull request looks good
